File: src/main/java/dynamodbpool/DynamoDBPool.java
package dynamodbpool;

import com.amazonaws.auth.AWSCredentialsProvider;
import com.amazonaws.regions.Region;
import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient;
import org.apache.commons.pool.impl.SoftReferenceObjectPool;

public class DynamoDBPool extends SoftReferenceObjectPool<AmazonDynamoDBClient> {
    public DynamoDBPool(AWSCredentialsProvider awsCredentialsProvider, Region region) {
        super(new DynamoDBClientFactory(awsCredentialsProvider, region));
    }
}

File: src/main/scala/dynamodbpool/DynamoDBPool.scala
package dynamodbpool

import org.apache.commons.pool.PoolableObjectFactory
import org.apache.commons.pool.impl.SoftReferenceObjectPool
import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient
import com.amazonaws.auth.{AWSCredentialsProvider, ClasspathPropertiesFileCredentialsProvider}
import com.amazonaws.regions.{Regions, Region}

class DynamoDBPool(val factory: PoolableObjectFactory[AmazonDynamoDBClient]) extends SoftReferenceObjectPool[AmazonDynamoDBClient](factory) {
  def this(awsCredentialsProvider: AWSCredentialsProvider, region: Region) =
    this(new DynamoDBClientPoolFactory(awsCredentialsProvider, region))
}

